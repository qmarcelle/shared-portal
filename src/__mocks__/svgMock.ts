import React from 'react';

const SvgMock = React.forwardRef<
  HTMLSpanElement,
  React.SVGProps<SVGSVGElement>
>((props, ref) => {
  return React.createElement('span', {
    ref,
    ...props,
    'data-testid': 'svg-mock',
  });
});

SvgMock.displayName = 'SvgMock';

export default SvgMock;
export const ReactComponent = SvgMock;
