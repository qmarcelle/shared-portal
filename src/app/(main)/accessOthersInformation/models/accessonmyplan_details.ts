export interface AccessOnMyPlanDetails {
  memberName: string;
  DOB: string;
  isOnline: boolean;
}
