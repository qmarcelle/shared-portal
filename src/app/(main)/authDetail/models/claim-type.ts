export enum ClaimType {
  Medical = 'Medical',
  Pharmacy = 'Pharmacy',
  Dental = 'Dental',
}
