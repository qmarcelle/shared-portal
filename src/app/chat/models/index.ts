// Export all chat-related types
export type {
  CobrowseConfig,
  CobrowseInitResponse,
  CobrowseSession,
  CobrowseSessionResponse,
} from './cobrowse';

// Make sure the ChatMessage type is consistent by defining it inline here
export interface ChatMessage {
  id: string;
  text: string;
  sender: 'user' | 'agent' | 'bot';
  timestamp: Date;
}

export * from './plans';
export type {
  ChatEligibility,
  ChatPayload,
  ChatSession,
  ChatSessionJWT,
  ChatState,
  PlanSwitcherState,
} from './session';
export type {
  BusinessDay,
  BusinessHours,
  ChatConfig,
  ChatOption,
  CobrowseState,
  UserEligibility,
} from './types';
