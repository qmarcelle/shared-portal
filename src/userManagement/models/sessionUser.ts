import { VisibilityRules } from '@/visibilityEngine/rules';

export enum UserRole {
  MEMBER = 'MEM',
  PERSONAL_REP = 'PR',
  AUTHORIZED_USER = 'AU',
  NON_MEM = 'NM',
}

export type SessionUser = {
  id: string;
  currUsr: {
    umpi: string;
    fhirId: string;
    role: UserRole;
    plan:
      | {
          memCk: string;
          sbsbCk: string;
          grgrCk: string;
          grpId: string;
          subId: string;
          fhirId: string;
          ntwkId: string;
        }
      | undefined;
  };
  rules?: string;
  vRules?: VisibilityRules;
  subscriberId?: string;
};
