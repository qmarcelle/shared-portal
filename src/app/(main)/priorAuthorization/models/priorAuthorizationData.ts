import { ClaimDetails } from '../../../../models/claim_details';

export const priorAuthorizationData: ClaimDetails[] = [
  {
    id: 'Claim1',
    claimStatus: 'Pending',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Physical Therapy',
    memberName: 'Chris Hall',
    serviceDate: '04/06/23',
    totalBilled: '',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Anand Patel',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim2',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Magnetic Resonance Imaging (MRI)',
    memberName: 'Chris Hall',
    serviceDate: '01/23/23',
    totalBilled: '145.00',
    planPaid: '114.76',
    myShare: '30.24',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim3',
    claimStatus: 'Partial Approval',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Continuous Positive Airway Pressure Machiâ€¦',
    memberName: 'Chris Hall',
    serviceDate: '08/10/23',
    totalBilled: '403.98',
    planPaid: '338.37',
    myShare: '65.61',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'Anand Patel',
    priorAuthFlag: true,
  },
  {
    id: 'Claim4',
    claimStatus: 'Denied',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Home Health Services',
    memberName: 'Chris Hall',
    serviceDate: '07/05/23',
    totalBilled: '403.98',
    planPaid: '0.00',
    myShare: '403.98',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'Home Health Services',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
  {
    id: 'Claim5',
    claimStatus: 'Approved',
    claimType: 'Medical',
    claimTotal: null,
    issuer: 'Brittant Johnson',
    memberName: 'Chris Hall',
    serviceDate: '07/04/23',
    totalBilled: '183.00',
    planPaid: '150.85',
    myShare: '32.15',
    claimsFlag: false,
    claimInfo: {},
    ReferredBy: 'N/A',
    ReferredTo: 'N/A',
    priorAuthFlag: true,
  },
];
