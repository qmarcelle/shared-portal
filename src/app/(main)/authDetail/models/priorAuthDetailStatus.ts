import { ClaimStatus } from '@/app/(main)/authDetail/models/claim-status';
import { ClaimType } from '@/app/(main)/authDetail/models/claim-type';

export interface PriorAuthDetailsStatus {
  priorAuthDetailType: ClaimType;
  priorAuthDetailName: string;
  dateOfVisit: string;
  priorAuthDetailStatus: ClaimStatus;
  member: string;
  PriorAuthReferenceId: string;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  authInfo: any;
  referredName: string;
}
