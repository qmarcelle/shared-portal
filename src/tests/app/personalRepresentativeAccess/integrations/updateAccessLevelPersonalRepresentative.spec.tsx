import PersonalRepresentativePage from '@/app/personalRepresentativeAccess/page';
import { AppModal, useAppModalStore } from '@/components/foundation/AppModal';
import { mockedAxios } from '@/tests/__mocks__/axios';
import { mockedFetch } from '@/tests/setup';
import { fetchRespWrapper } from '@/tests/test_utils';
import { UserRole } from '@/userManagement/models/sessionUser';
import '@testing-library/jest-dom';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';

const vRules = {
  user: {
    id: 'Testuser553',
    currUsr: {
      firstName: 'Chris',
      role: UserRole.MEMBER,
      plan: {
        planName: 'BlueCross BlueShield of Tennessee',
        subId: '123456',
        grpId: '100000',
        memCk: '147235702',
        coverageType: ['Medical', 'Dental', 'Vision'],
      },
    },
    vRules: {
      active: true,
      matureMinor: true,
    },
  },
};
jest.mock('src/auth', () => ({
  auth: jest.fn(),
}));
const renderUI = async () => {
  const component = await PersonalRepresentativePage();
  const { container } = render(
    <>
      <AppModal />
      {component}
    </>,
  );

  return container;
};

jest.mock('src/auth', () => ({
  auth: jest.fn(),
}));

describe('Updating Level of Access for Personal Representative', () => {
  beforeEach(() => {
    const { dismissModal } = useAppModalStore.getState();
    dismissModal();
  });
  it('API Success Scenario', async () => {
    const mockAuth = jest.requireMock('src/auth').auth;
    vRules.user.vRules.active = true;
    vRules.user.vRules.matureMinor = true;
    vRules.user.currUsr.role = UserRole.MEMBER;
    mockAuth.mockResolvedValue(vRules);

    mockedFetch.mockResolvedValueOnce(
      fetchRespWrapper({
        data: {
          getPBEmessage: 'Person Record fetched Successfully ',
          getConsentmessage: 'Person Record fetched Successfully ',
          getPBECorrelationId: '0124b43b-efd0-42bc-b904-cb2ac9c712b9',
          consentCorrelationId: 'cd30f2b3-3bae-42f3-b93a-8fa42f3884e7',
          getPBEDetails: [
            {
              umpid: '57c85test3ebd23c7db88245',
              userName: 'Testuser553',
              personFHIRID: '30345928-abcd-ef01-2345-6789abcdef51',
              firstName: 'Quino',
              middleName: 'S',
              lastName: 'Deper',
              suffix: 'Mr.',
              address1: 'street1',
              address2: 'street2',
              phoneNumber: '8095469997',
              city: 'Ongole',
              state: 'AP',
              zip: 34566,
              dob: '2019-10-29T00:00:10.0000000+00:00',
              gender: 'M',
              email: 'sub.m@gmail.com',
              relationshipInfo: [
                {
                  personRoleType: 'Dependent',
                  org: 'bcbst_facets',
                  roleTermDate: '2030-11-30T00:00:00.0000000+00:00',
                  nativeId: '38922455201-100000',
                  primaryPlanFlag: false,
                  patientFHIRID: '30345928-abcd-ef01-2345-6789abcdef52',
                  userName: 'Testuser553',
                  memeCk: '147235702',
                  clientId: '194',
                  multiPlanConfirmed: false,
                  multiPlanConfirmedDate: '2030-11-30T00:00:00.0000000+00:00',
                  approvalRequestId: '38922455201-1123456787',
                  relatedPersons: [
                    {
                      relatedPersonUMPID: '57c85test3ebd23c7db88244',
                      relatedPersonFirstName: 'Chris',
                      relatedPersonLastName: 'Hall',
                      relatedPersonMiddleName: 'S',
                      relatedPersonSuffix: 'Mr.',
                      relatedPersonNativeId: '38922455200-100000',
                      relatedPersonFHIRID: '',
                      relatedPersonPatientFHIRID: '',
                      relatedPersonRelationshipTermDate:
                        '2030-11-30T00:00:00.0000000+00:00',
                      relatedPersonRoleType: 'PR',
                      relatedPersonDob: '1964-07-20',
                      relatedPersonApprovalRequestId: '',
                      relatedPersonMemeCk: '6765454',
                      id: 'e7d4e7e9-5f98-4e35-9141-de8c8eb944a6',
                      effectiveOn: '2024-11-11T05:00:00Z',
                      expiresOn: '2026-11-11T05:00:00Z',
                      policyId: 'out-of-box-policy-0002',
                    },
                  ],
                },
              ],
            },
          ],
          consentDetails: [
            {
              id: 'e7d4e7e9-5f98-4e35-9141-de8c8eb944a6',
              name: 'Consent - Limited Access',
              description:
                'This consent is governed by limited access policy for HIV and Cancer information',
              status: 'active',
              performer: 'john.doe@bcbst.com',
              requester: 'Person/24baffdb-76b7-4cb1-bf97-47f4bcf5574a',
              requestee: 'Patient/e3c212e5-1b7f-4521-8ba1-3b48befa98d6',
              policyId: 'ce3ee5b4-3810-4c8a-bfd9-9d8681853a7e',
              policyBusinessIdentifier: 'pol_10237_lim_ac',
              type: 'PEOPLE_TO_PEOPLE',
              effectiveOn: '2023-01-01T12:00:00Z',
              expiresOn: '2023-12-31T12:00:00Z',
              signedOn: '2023-01-01T12:00:00Z',
              modifiedAt: '2023-01-01T12:00:00Z',
              accessControl: 'PERMIT_ALL',
              options: [
                {
                  id: 'f83a70bf-736d-4dfa-96eb-31d5c5a09535',
                  businessIdentifier: 'option_biz_id_001',
                  caption: 'HIV',
                  mappings: [
                    {
                      criteria: 'CODE',
                      key: 'http://hl7.org/fhir/sid/icd-10',
                      value: 'B20,B9735',
                    },
                  ],
                },
              ],
            },
          ],
        },
        details: {
          componentName: 'string',
          componentStatus: 'string',
          returnCode: 'string',
          subSystemName: 'string',
          message: 'string',
          problemTypes: 'string',
          innerDetails: {
            statusDetails: [
              {
                componentName: 'string',
                componentStatus: 'string',
                returnCode: 'string',
                subSystemName: 'string',
                message: 'string',
                problemTypes: 'string',
                innerDetails: {
                  statusDetails: ['string'],
                },
              },
            ],
          },
        },
      }),
    );

    const component = await renderUI();
    expect(mockedFetch).toHaveBeenCalledWith(
      'ES_SVC_URL/searchMemberLookupDetails/getPBEConsentDetails?userName=Testuser553&isPBERequired=true&isConsentRequired=true',
      {
        cache: undefined,
        headers: { Authorization: 'Bearer BearerTokenMockedValue' },
        next: { revalidate: 1800, tags: ['Testuser553'] },
      },
    );
    expect(screen.getByText('Personal Representative Access')).toBeVisible();
    expect(screen.getByText('Chris Hall')).toBeVisible();
    expect(screen.getByText('DOB: 07/20/1964')).toBeVisible();
    expect(screen.getByText('Basic Access as of 01/01/2024')).toBeVisible();

    expect(screen.getByText('Update')).toBeVisible();

    fireEvent.click(screen.getByText('Update'));

    expect(
      screen.getByRole('heading', { name: 'Edit Level Of Access' }),
    ).toBeVisible();

    expect(screen.getByText('Edit Level Of Access')).toBeVisible();

    fireEvent.click(screen.getByRole('button', { name: /Next/i }));
    expect(screen.getByText('Save Permissions')).toBeVisible();
    mockedAxios.get.mockResolvedValueOnce({
      data: {
        data: {
          message: 'Success',
        },
      },
    });
    fireEvent.click(screen.getByRole('button', { name: /Save Permissions/i }));

    await waitFor(() => {
      expect(mockedAxios.post).toHaveBeenCalledWith(
        '/consentOperations/updateConsent?consentId=e7d4e7e9-5f98-4e35-9141-de8c8eb944a6',
        {
          effectiveOn: '2024-11-11T05:00:00Z',
          expiresOn: '2026-11-11T05:00:00Z',
          policyId: 'out-of-box-policy-0002',
          requestType: 'update',
        },
      );
    });

    expect(component).toMatchSnapshot();
  });
  it('API Failure Scenario', async () => {
    const mockAuth = jest.requireMock('src/auth').auth;
    vRules.user.vRules.active = true;
    vRules.user.vRules.matureMinor = true;
    vRules.user.currUsr.role = UserRole.MEMBER;
    mockAuth.mockResolvedValue(vRules);

    mockedFetch.mockResolvedValueOnce(
      fetchRespWrapper({
        data: {
          getPBEmessage: 'Person Record fetched Successfully ',
          getConsentmessage: 'Person Record fetched Successfully ',
          getPBECorrelationId: '0124b43b-efd0-42bc-b904-cb2ac9c712b9',
          consentCorrelationId: 'cd30f2b3-3bae-42f3-b93a-8fa42f3884e7',
          getPBEDetails: [
            {
              umpid: '57c85test3ebd23c7db88245',
              userName: 'Testuser553',
              personFHIRID: '30345928-abcd-ef01-2345-6789abcdef51',
              firstName: 'Quino',
              middleName: 'S',
              lastName: 'Deper',
              suffix: 'Mr.',
              address1: 'street1',
              address2: 'street2',
              phoneNumber: '8095469997',
              city: 'Ongole',
              state: 'AP',
              zip: 34566,
              dob: '2019-10-29T00:00:10.0000000+00:00',
              gender: 'M',
              email: 'sub.m@gmail.com',
              relationshipInfo: [
                {
                  personRoleType: 'Dependent',
                  org: 'bcbst_facets',
                  roleTermDate: '2030-11-30T00:00:00.0000000+00:00',
                  nativeId: '38922455201-100000',
                  primaryPlanFlag: false,
                  patientFHIRID: '30345928-abcd-ef01-2345-6789abcdef52',
                  userName: 'Testuser553',
                  memeCk: '147235702',
                  clientId: '194',
                  multiPlanConfirmed: false,
                  multiPlanConfirmedDate: '2030-11-30T00:00:00.0000000+00:00',
                  approvalRequestId: '38922455201-1123456787',
                  relatedPersons: [
                    {
                      relatedPersonUMPID: '57c85test3ebd23c7db88244',
                      relatedPersonFirstName: 'Chris',
                      relatedPersonLastName: 'Hall',
                      relatedPersonMiddleName: 'S',
                      relatedPersonSuffix: 'Mr.',
                      relatedPersonNativeId: '38922455200-100000',
                      relatedPersonFHIRID: '',
                      relatedPersonPatientFHIRID: '',
                      relatedPersonRelationshipTermDate:
                        '2030-11-30T00:00:00.0000000+00:00',
                      relatedPersonRoleType: 'PR',
                      relatedPersonDob: '1964-07-20',
                      relatedPersonApprovalRequestId: '',
                      relatedPersonMemeCk: '6765454',
                      id: 'e7d4e7e9-5f98-4e35-9141-de8c8eb944a6',
                    },
                  ],
                },
              ],
            },
          ],
          consentDetails: [
            {
              id: 'e7d4e7e9-5f98-4e35-9141-de8c8eb944a6',
              name: 'Consent - Limited Access',
              description:
                'This consent is governed by limited access policy for HIV and Cancer information',
              status: 'active',
              performer: 'john.doe@bcbst.com',
              requester: 'Person/24baffdb-76b7-4cb1-bf97-47f4bcf5574a',
              requestee: 'Patient/e3c212e5-1b7f-4521-8ba1-3b48befa98d6',
              policyId: 'ce3ee5b4-3810-4c8a-bfd9-9d8681853a7e',
              policyBusinessIdentifier: 'pol_10237_lim_ac',
              type: 'PEOPLE_TO_PEOPLE',
              effectiveOn: '2023-01-01T12:00:00Z',
              expiresOn: '2023-12-31T12:00:00Z',
              signedOn: '2023-01-01T12:00:00Z',
              modifiedAt: '2023-01-01T12:00:00Z',
              accessControl: 'PERMIT_ALL',
              options: [
                {
                  id: 'f83a70bf-736d-4dfa-96eb-31d5c5a09535',
                  businessIdentifier: 'option_biz_id_001',
                  caption: 'HIV',
                  mappings: [
                    {
                      criteria: 'CODE',
                      key: 'http://hl7.org/fhir/sid/icd-10',
                      value: 'B20,B9735',
                    },
                  ],
                },
              ],
            },
          ],
        },
        details: {
          componentName: 'string',
          componentStatus: 'string',
          returnCode: 'string',
          subSystemName: 'string',
          message: 'string',
          problemTypes: 'string',
          innerDetails: {
            statusDetails: [
              {
                componentName: 'string',
                componentStatus: 'string',
                returnCode: 'string',
                subSystemName: 'string',
                message: 'string',
                problemTypes: 'string',
                innerDetails: {
                  statusDetails: ['string'],
                },
              },
            ],
          },
        },
      }),
    );

    const component = await renderUI();
    expect(mockedFetch).toHaveBeenCalledWith(
      'ES_SVC_URL/searchMemberLookupDetails/getPBEConsentDetails?userName=Testuser553&isPBERequired=true&isConsentRequired=true',
      {
        cache: undefined,
        headers: { Authorization: 'Bearer BearerTokenMockedValue' },
        next: { revalidate: 1800, tags: ['Testuser553'] },
      },
    );
    expect(screen.getByText('Personal Representative Access')).toBeVisible();
    expect(screen.getByText('Chris Hall')).toBeVisible();
    expect(screen.getByText('DOB: 07/20/1964')).toBeVisible();
    expect(screen.getByText('Basic Access as of 01/01/2024')).toBeVisible();

    expect(screen.getByText('Update')).toBeVisible();

    fireEvent.click(screen.getByText('Update'));

    expect(
      screen.getByRole('heading', { name: 'Edit Level Of Access' }),
    ).toBeVisible();

    expect(screen.getByText('Edit Level Of Access')).toBeVisible();

    fireEvent.click(screen.getByRole('button', { name: /Next/i }));
    expect(screen.getByText('Save Permissions')).toBeVisible();
    mockedAxios.get.mockResolvedValueOnce({
      data: {
        data: {
          message: 'Failure',
        },
      },
    });
    fireEvent.click(screen.getByRole('button', { name: /Save Permissions/i }));

    await waitFor(() => {
      expect(mockedAxios.post).toHaveBeenCalledWith(
        '/consentOperations/updateConsent?consentId=e7d4e7e9-5f98-4e35-9141-de8c8eb944a6',
        {
          effectiveOn: '2024-11-11T05:00:00Z',
          expiresOn: '2026-11-11T05:00:00Z',
          policyId: 'out-of-box-policy-0002',
          requestType: 'update',
        },
      );
    });

    await waitFor(() => {
      expect(
        screen.getByRole('heading', { name: 'Try Again Later' }),
      ).toBeVisible();
    });

    expect(component).toMatchSnapshot();
  });
});
