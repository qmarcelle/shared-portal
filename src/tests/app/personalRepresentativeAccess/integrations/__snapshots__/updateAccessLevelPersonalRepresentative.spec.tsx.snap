// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Updating Level of Access for Personal Representative API Failure Scenario 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center page"
  >
    <div
      class="flex flex-col app-content app-base-font-color"
    >
      <span
        style="display: block; width: 1px; min-width: 1px; height: 52px; min-height: 52px;"
      />
      <h1
        class="title-1 pl-3"
      >
        Personal Representative Access
      </h1>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
      />
      <p
        class="body-1 w-2/3 ml-4"
        style="display: block;"
      >
        Personal representatives have the legal authority to make health care decisions on behalf of the member.
      </p>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 22px; min-height: 22px;"
      />
      <section
        class="flex flex-row items-start app-body"
      >
        <div
          class="flex flex-col flex-grow page-section-36_67 items-stretch"
        >
          <div
            class="card-main  large-section"
          >
            <div
              class="flex flex-col items-stretch"
            >
              <div>
                <h1
                  class="title-1 title-2"
                >
                  Understanding Access
                </h1>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 22px; min-height: 22px;"
                />
              </div>
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <p
                    class="mr-2 flex-grow"
                  >
                    Full Access
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                />
              </div>
              <div
                class=""
                style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
              />
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <p
                    class="mr-2 flex-grow"
                  >
                    Basic Access
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                />
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="flex flex-col page-section-63_33 items-stretch"
        >
          <div
            class="card-main large-section"
          >
            <div>
              <div
                class="flex flex-col flex flex-col"
              >
                <h2
                  class="title-2"
                >
                  Your Representative(s)
                </h2>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="flex flex-col flex flex-col"
                >
                  <div
                    class="card-elevated cursor-pointer mb-4"
                  >
                    <div
                      class="flex flex-col m-8"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-row justify-between"
                      >
                        <p
                          class="body-1 font-bold body-1"
                          style="display: block;"
                        >
                          Chris Hall
                        </p>
                        <p
                          class="body-1"
                          style="display: block;"
                        >
                          DOB: 07/20/1964
                        </p>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class=""
                        style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                      />
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-col"
                      >
                        <div
                          class="flex flex-row"
                        >
                          <span
                            style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
                          />
                        </div>
                        <div>
                          <div
                            class="flex flex-row"
                          >
                            <p
                              class="body-1 ml-2"
                              style="display: block;"
                            >
                              Basic Access as of 01/01/2024
                            </p>
                            <span
                              style="display: block; width: 1px; min-width: 1px; height: 42px; min-height: 42px;"
                            />
                          </div>
                          <div
                            class="flex flex-row"
                          >
                            <span
                              style="display: block; width: 1px; min-width: 1px; height: 42px; min-height: 42px;"
                            />
                            <div
                              class="flex flex-row"
                            >
                              <p
                                class="font-bold primary-color ml-2"
                              >
                                Update
                              </p>
                              <div
                                class="ml-2"
                              >
                                <img
                                  alt="link"
                                  data-nimg="1"
                                  decoding="async"
                                  height="40"
                                  loading="lazy"
                                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                                  style="color: transparent;"
                                  width="40"
                                />
                              </div>
                            </div>
                          </div>
                          <div
                            class="flex flex-row"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                />
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <span
                  class="rich-text body-2 undefined"
                >
                  <span>
                    For assistance, please
                  </span>
                  <span
                    class="link"
                  >
                    <a>
                       start a chat 
                    </a>
                  </span>
                  <span>
                    or call us at [1-800-000-000]
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div />
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`Updating Level of Access for Personal Representative API Success Scenario 1`] = `
<div>
  <div
    class="flex flex-col justify-center items-center page"
  >
    <div
      class="flex flex-col app-content app-base-font-color"
    >
      <span
        style="display: block; width: 1px; min-width: 1px; height: 52px; min-height: 52px;"
      />
      <h1
        class="title-1 pl-3"
      >
        Personal Representative Access
      </h1>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
      />
      <p
        class="body-1 w-2/3 ml-4"
        style="display: block;"
      >
        Personal representatives have the legal authority to make health care decisions on behalf of the member.
      </p>
      <span
        style="display: block; width: 1px; min-width: 1px; height: 22px; min-height: 22px;"
      />
      <section
        class="flex flex-row items-start app-body"
      >
        <div
          class="flex flex-col flex-grow page-section-36_67 items-stretch"
        >
          <div
            class="card-main  large-section"
          >
            <div
              class="flex flex-col items-stretch"
            >
              <div>
                <h1
                  class="title-1 title-2"
                >
                  Understanding Access
                </h1>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 22px; min-height: 22px;"
                />
              </div>
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <p
                    class="mr-2 flex-grow"
                  >
                    Full Access
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                />
              </div>
              <div
                class=""
                style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
              />
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4 "
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <p
                    class="mr-2 flex-grow"
                  >
                    Basic Access
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                />
              </div>
            </div>
          </div>
          <div />
        </div>
        <div
          class="flex flex-col page-section-63_33 items-stretch"
        >
          <div
            class="card-main large-section"
          >
            <div>
              <div
                class="flex flex-col flex flex-col"
              >
                <h2
                  class="title-2"
                >
                  Your Representative(s)
                </h2>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <div
                  class="flex flex-col flex flex-col"
                >
                  <div
                    class="card-elevated cursor-pointer mb-4"
                  >
                    <div
                      class="flex flex-col m-8"
                    >
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-row justify-between"
                      >
                        <p
                          class="body-1 font-bold body-1"
                          style="display: block;"
                        >
                          Chris Hall
                        </p>
                        <p
                          class="body-1"
                          style="display: block;"
                        >
                          DOB: 07/20/1964
                        </p>
                      </div>
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class=""
                        style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                      />
                      <span
                        style="display: block; width: 1px; min-width: 1px; height: 16px; min-height: 16px;"
                      />
                      <div
                        class="flex flex-col"
                      >
                        <div
                          class="flex flex-row"
                        >
                          <span
                            style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
                          />
                        </div>
                        <div>
                          <div
                            class="flex flex-row"
                          >
                            <p
                              class="body-1 ml-2"
                              style="display: block;"
                            >
                              Basic Access as of 01/01/2024
                            </p>
                            <span
                              style="display: block; width: 1px; min-width: 1px; height: 42px; min-height: 42px;"
                            />
                          </div>
                          <div
                            class="flex flex-row"
                          >
                            <span
                              style="display: block; width: 1px; min-width: 1px; height: 42px; min-height: 42px;"
                            />
                            <div
                              class="flex flex-row"
                            >
                              <p
                                class="font-bold primary-color ml-2"
                              >
                                Update
                              </p>
                              <div
                                class="ml-2"
                              >
                                <img
                                  alt="link"
                                  data-nimg="1"
                                  decoding="async"
                                  height="40"
                                  loading="lazy"
                                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                                  style="color: transparent;"
                                  width="40"
                                />
                              </div>
                            </div>
                          </div>
                          <div
                            class="flex flex-row"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 12px; min-height: 12px;"
                />
                <div
                  class=""
                  style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
                />
                <span
                  style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
                />
                <span
                  class="rich-text body-2 undefined"
                >
                  <span>
                    For assistance, please
                  </span>
                  <span
                    class="link"
                  >
                    <a>
                       start a chat 
                    </a>
                  </span>
                  <span>
                    or call us at [1-800-000-000]
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div />
        </div>
      </section>
    </div>
  </div>
</div>
`;
